RIOT-OS/RIOT
# cpu/esp*: code deduplication and cleanup

### Contribution description

After PR #11108 has been merged, it became possible to deduplicate a lot of the ESP32 and ESP8266 code by moving it to `cpu/esp_common`. This PR realizes these deduplications and some cleanup that are tracked in issue #10658.

### Testing procedure

`examples/gnrc_networking` can be used as test that includes most parts of the changes. This should be tested with any ESP8266 and any ESP32 board.
```
USEMODULE='esp_wifi' CFLAGS='-DESP_WIFI_SSID=\"ssid\" -DESP_WIFI_PASS=\"pass\"' \
make BOARD=esp... -C examples/gnrc_networking flash term
```
Furthermore: changed peripherals should to be tested.
- flash: ```make BOARD=esp... -C tests/pkg_spiffs flash test```
- i2c: ```make BOARD=esp... -C tests/periph_i2c flash term```
- hwrng: ```make BOARD=esp... -C tests/periph_hwrng flash test```
- spi: ```make BOARD=esp... -C tests/periph_spi flash term```
- uart: ```make BOARD=esp... -C tests/periph_uart flash term``` (only possible on ESP32)

### Issues/PRs references

Tracked in issue #10658
Fixes issue #11571 
Contains the changes of PRs #12967, #13004
Depends on #13035
